big-o-notation/big-o.js

```diff
@@ -1,44 +1,44 @@
 /* exported uniqueQuadratic, uniqueLinear */
 
 function uniqueLinear(words) {
-  const seen = {};                  // _ * _ = O(?)
-  const unique = [];                // _ * _ = O(?)
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
   for (
-    let i = 0;                      // _ * _ = O(?)
-    i < words.length;               // _ * _ = O(?)
-    i++                             // _ * _ = O(?)
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= O(n)
+    i++                             // 2 * n = O(2n) ~= O(n)
   ) {
-    const word = words[i];          // _ * _ = O(?)
-    if (!seen[word]) {              // _ * _ = O(?)
-      seen[word] = true;            // _ * _ = O(?)
-      unique[unique.length] = word; // _ * _ = O(?)
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    if (!seen[word]) {              // 2 * n = O(2n) ~= O(n)
+      seen[word] = true;            // 2 * n = O(2n) ~= O(n)
+      unique[unique.length] = word; // 3 * n = O(3n) ~= O(n)
     }
   }
-  return unique;                    // _ * _ = O(?)
-} // Big O Notation for uniqueLinear: O(?)
+  return unique;                    // 1 * 1 = O(1)
+} // Big O Notation for uniqueLinear: O(n)
 
 function uniqueQuadratic(words) {
-  const unique = [];                // _ * _ = O(?)
+  const unique = [];                // 1 * 1 = O(1)
   for (
-    let i = 0;                      // _ * _ = O(?)
-    i < words.length;               // _ * _ = O(?)
-    i++                             // _ * _ = O(?)
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= O(n)
+    i++                             // 2 * n = O(2n) ~= O(n)
   ) {
-    const word = words[i];          // _ * _ = O(?)
-    let isUnique = true;            // _ * _ = O(?)
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    let isUnique = true;            // 1 * 1 = O(1)
     for (
-      let c = 0;                    // _ * _ = O(?)
-      c < i;                        // _ * _ = O(?)
-      c++                           // _ * _ = O(?)
+      let c = 0;                    // 1 * n     = O(n) NESTED LOOP!
+      c < i;                        // 2 * n * n = O(2n^2) ~= O(n^2)
+      c++                           // 2 * n * n = O(2n^2) ~= O(n^2)
     ) {
-      const comparing = words[c];   // _ * _ = O(?)
-      if (comparing === word) {     // _ * _ = O(?)
-        isUnique = false;           // _ * _ = O(?)
+      const comparing = words[c];   // 2 * n = O(2n) ~= O(n)
+      if (comparing === word) {     // 1 * n * n = O(n^2)
+        isUnique = false;           // 1 * n = O(n)
```

Comment:
This one should be n^2. When doing complexity analysis, assume every branch is taken every time.